handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslContext.java

```diff
@@ -642,8 +642,8 @@ static void setKeyMaterial(long ctx, X509Certificate[] keyCertChain, PrivateKey
         try {
             // Only encode one time
             encoded = PemX509Certificate.toPEM(ByteBufAllocator.DEFAULT, true, keyCertChain);
-            keyCertChainBio = newBIO(encoded.content().retainedSlice());
-            keyCertChainBio2 = newBIO(encoded.content().retainedSlice());
+            keyCertChainBio = toBIO(ByteBufAllocator.DEFAULT, encoded.retainedDuplicate());
+            keyCertChainBio2 = toBIO(ByteBufAllocator.DEFAULT, encoded.retainedDuplicate());
```

Comment:
Are the `#retainedDuplicate()` calls correct? Inside the `toBIO()` method its doing an another round of retained slicing.